compiler/rustc_errors/src/lib.rs

```diff
@@ -983,7 +991,14 @@ impl<'a> DiagCtxtHandle<'a> {
     }
 
     pub fn has_stashed_diagnostic(&self, span: Span, key: StashKey) -> bool {
-        self.inner.borrow().stashed_diagnostics.get(&(span.with_parent(None), key)).is_some()
+        let inner = self.inner.borrow();
```

Comment:
You may wonder... is it sufficient here to just call `self.inner.borrow().stashed_diagnostics.is_empty()` as a fast path, b/c we obviously should never have stashed diagnostics on the good path?

Well no! We stash `AssociatedTypeSuggestion` in a bunch of places on the good path! What the heck, lol. I'll open a separate PR to gauge the perf impact of that.